runtime/j9vm/java9vmi.c

```diff
@@ -951,93 +952,87 @@ JVM_DefineModule(JNIEnv * env, jobject module, jstring version, jstring location
 			vmFuncs->setCurrentExceptionUTF(currentThread, J9VMCONSTANTPOOL_JAVALANGILLEGALARGUMENTEXCEPTION, "module is unnamed");
 		} else if (!isModuleNameValid(moduleName)) {
 			vmFuncs->setCurrentExceptionUTF(currentThread, J9VMCONSTANTPOOL_JAVALANGILLEGALARGUMENTEXCEPTION, "module name is invalid");
-		} else if (NULL != classLoader) {
-			BOOLEAN success = FALSE;
-			J9Module *j9mod = createModule(currentThread, modObj, classLoader, moduleName);
-			UDATA rc = ERRCODE_GENERAL_FAILURE;
-
-			if (NULL != j9mod) {
+		} else if (NULL == classLoader) {
+			/* An exception should be pending if classLoader is null */
+			assert(NULL != currentThread->currentException);
```

Comment:
This should be a trace assert, not a C assert.  (I think this is pre-existing code but we should fix it while we're here)

Reply:
Fixed.